include/operations/blas_operators.hpp

```diff
@@ -40,22 +41,19 @@ namespace blas {
 template <typename ScalarT>
 struct IndVal {
   using value_type = ScalarT;
-  size_t ind;
-  value_type val;
+  size_t ind = UINT_MAX;
```

Comment:
Is there any reason that you set the value here and next line instead of using the constructor to assign a value?